tests/test_datasets/test_internal_datasets.py

```diff
@@ -136,3 +164,58 @@ def test_dataset_statistics(dataset_name, expected_shape, expected_min_date, exp
     assert ts_full.index.min() == expected_min_date
     assert ts_full.index.max() == expected_max_date
     assert ts_full.df.shape[0] == parts_rows
+
+
+@pytest.mark.parametrize(
+    "dataset_name, expected_train_df_exog_shape, expected_test_df_exog_shape, expected_train_df_exog_dates, expected_test_df_exog_dates, dataset_parts",
+    [
+        (
+            "m3_monthly",
+            (126 + 18, 1428),
+            (18, 1428),
+            (pd.to_datetime("2010-01-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            (pd.to_datetime("2020-07-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            ("train", "test"),
+        ),
+        (
+            "m3_quarterly",
+            (64 + 8, 756),
+            (8, 756),
+            (pd.to_datetime("2004-03-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            (pd.to_datetime("2020-03-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            ("train", "test"),
+        ),
+        (
+            "m3_other",
+            (96 + 8, 174),
+            (8, 174),
+            (pd.to_datetime("1996-03-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            (pd.to_datetime("2020-03-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            ("train", "test"),
+        ),
+        (
+            "m3_yearly",
+            (41 + 6, 645),
+            (6, 645),
+            (pd.to_datetime("1975-12-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            (pd.to_datetime("2016-12-31 00:00:00"), pd.to_datetime("2021-12-31 00:00:00")),
+            ("train", "test"),
+        ),
+    ],
+)
+def test_df_exog_statistics(
+    dataset_name,
+    expected_train_df_exog_shape,
```

Comment:
This isn't a really robust parametrization. Let's add parameters like `expected_exog_shapes`, `expected_dates` there we store values for each part.